data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DefaultLogstashPluginAttributesMapper.java

```diff
@@ -12,18 +12,28 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 class DefaultLogstashPluginAttributesMapper implements LogstashPluginAttributesMapper {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultLogstashPluginAttributesMapper.class);
 
     @Override
     public Map<String, Object> mapAttributes(final List<LogstashAttribute> logstashAttributes, final LogstashAttributesMappings logstashAttributesMappings) {
-        final Map<String, Object> pluginSettings = new LinkedHashMap<>(logstashAttributesMappings.getAdditionalAttributes());
 
+        Objects.requireNonNull(logstashAttributes);
+        Objects.requireNonNull(logstashAttributesMappings);
+
+        final Map<String, Object> pluginSettings = new LinkedHashMap<>();
+
+        if(logstashAttributesMappings.getAdditionalAttributes() != null) {
```

Comment:
Would it be feasible and nicer to have logstashAttributesMappings.getAdditionalAttributes() already return a non-null value?  If it returns an empty map in this case, it would be great.

Reply:
Yea, I'd like that.
The underlying cause is that the model is being deserialized from YAML. So, we will need to make the model return an empty list instead of `null` when it is not set. But, this should be possible.